documentation/examples/simpleResourceSample.json

```diff
@@ -0,0 +1,307 @@
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Accepted"
+          },
+          "204": {
+            "description": "No Content"
+          }
+        },
+        "x-ms-long-running-operation": true
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.SwaggerSample/sampleResources": {
+      "get": {
+        "tags": [
+          "SwaggerSample"
+        ],
+        "operationId": "SampleResources_ListByResourceGroup",
+        "description": "Delete an instance of your sample resource",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The list of sample resources in the given resource group",
+            "schema": {
+              "$ref": "#/definitions/SampleResourceList"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.SwaggerSample/sampleResources": {
+      "get": {
+        "tags": [
+          "SwaggerSample"
+        ],
+        "operationId": "SampleResources_List",
+        "description": "List all sampleResource instances in the given subscription",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The list of sample resources in the given subscription",
+            "schema": {
+              "$ref": "#/definitions/SampleResourceList"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Resource": {
+      "description": "Common resource definition. TODO: move to external reference/x-swagger common definition",
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource Id"
+        },
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource name"
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource type"
+        },
+        "location": {
+          "type": "string",
+          "description": "Resource location"
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags"
+        }
+      },
+      "required": [
```

Comment:
We should consider moving the required:ness out of the base resource since a) it is not required for all resource types and b) it makes the model definition impossible to use in PATCH scenarios.